babel.config.js

```diff
@@ -160,3 +164,43 @@ module.exports = function (api) {
 
   return config;
 };
+
+// TODO(Babel 8) This polyfill is only needed for Node.js 6 and 8
+function pluginPolyfillRequireResolve({ template, types: t }) {
+  return {
+    visitor: {
+      MemberExpression(path) {
+        if (!path.matchesPattern("require.resolve")) return;
+        if (!t.isCallExpression(path.parent, { callee: path.node })) return;
+
+        const args = path.parent.arguments;
+        if (args.length < 2) return;
+        if (
+          !t.isObjectExpression(args[1]) ||
+          args[1].properties.length !== 1 ||
+          !t.isIdentifier(args[1].properties[0].key, { name: "paths" }) ||
+          !t.isArrayExpression(args[1].properties[0].value) ||
+          args[1].properties[0].value.elements.length !== 1
+        ) {
+          throw path.parentPath.buildCodeFrameError(
+            "This 'require.resolve' usage is not supported by the inline polyfill."
+          );
+        }
+
+        // require.resolve's paths option has been introduced in Node.js 8.9
+        // https://nodejs.org/api/modules.html#modules_require_resolve_request_options
+        path.replaceWith(template.ast`
+          parseFloat(process.versions.node) >= 8.9
+            ? require.resolve
+            : (/* request */ r, { paths: [/* base */ b] }, M = require("module")) => {
+                let /* filename */ f = M._findPath(r, M._nodeModulePaths(b).concat(b));
+                if (f) return f;
+                f = new Error(\`Cannot resolve module '\${r}'\`);
```

Comment:
```suggestion
                f = new Error(\`Cannot find module '\${r}'\`);
```
To be consistent with the native `require.resolve` error message.

Reply:
Actually, I put `resolve` here otherwise all the tests where failing. I think Jest's `require.resolve` uses `"resolve"` in its error message.

Reply:
Not a blocker.

@SimenB Is it possible to test MODULE_NOT_FOUND error without changing the error message? We want to make assertion about the MODULE_NOT_FOUND error but Jest seems to assume such error means the test can not be run and all tests are failing.

Reply:
error from jest should match node, if it doesn't it's a bug

Reply:
@JLHwung The jest version we are using on Node 6 didn't always set `MODULE_NOT_FOUND` anyway, so we can't test it right now.

Reply:
I don't really follow here, but if Jest 26 does something wrong with `MODULE_NOT_FOUND` it would be great if you opened up an issue 🙂

Reply:
No don't worry! We are using Jest 24 on Node.js 6 and Jest 26 where it's supported. The bug was only with Jest 24, so we cannot rely on the fix but it is fixed in Jest 26!

Reply:
ah, good stuff 👍